### PR TITLE
new package: syslinux 6.0.4-pre1

### PR DESCRIPTION
**Pull request by a-schaefers**
## Description of package
Basically follows alpine build to a t + adds some nice things from void
## Checklist

- [x] Latest upstream version.
- [x] I agree to maintain this package (**required**).
- [x] I agree to notify KISS if I can no longer maintain this package (**required**).

### CODE REVIEW

**Comment by a-schaefers** [comment]
Alpine's "update-extlinux" scripts I am not a fan and removed it. Extlinux is stupid simple.

My grub to syslinux conversion (using BIOS with mbr partition table) was this easy:

kiss i syslinux
```sh
dd bs=440 count=1 conv=notrunc if=/usr/share/syslinux/mbr.bin of=/dev/sda
extlinux --install /boot --device=/dev/sda1 # device should be the partition on which /boot is mounted
cat << EOF > /boot/extlinux.conf
DEFAULT menu.c32
PROMPT 0
MENU TITLE Linux Boot Menu
MENU HIDDEN
MENU AUTOBOOT Linux will be booted automatically in # seconds.
TIMEOUT 30
LABEL 0
  MENU LABEL Linux
  LINUX vmlinuz
  APPEND root=PARTUUID=1af08dd6-01 modules=sd-mod,usb-storage,ext4 ro init=/boot/bin/sinit quiet

MENU SEPARATOR
EOF # note that since we do not use initrd the INITRD line has been removed
```

File: community/syslinux/build [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh -e
+
+unset CFLAGS LDFLAGS
```
Comment (dylanaraps) [inline community/syslinux/build]: Any reason for this?

File: community/syslinux/build [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# apply patches
+for f in *.patch ; do patch -p1 < "$f"; done
```
Comment (dylanaraps) [inline community/syslinux/build]:
```sh
patch -p1 < fix-sysmacros.patch
patch -p1 < 0018-prevent-pow-optimization.patch
```

Comment (dylanaraps) [reply community/syslinux/build]: Remove the comment too. :+1:

File: community/syslinux/patches/0018-prevent-pow-optimization.patch [hunk 3]
```diff
@@ -0,0 +1,36 @@
+From: Lukas Schwaighofer <lukas@schwaighofer.name>
```
Comment (dylanaraps) [inline community/syslinux/patches/0018-prevent-pow-optimization.patch]: This looks like a much nicer patch: https://git.archlinux.org/svntogit/packages.git/tree/trunk/0018-prevent-pow-optimization.patch?h=packages/syslinux

File: community/syslinux/build [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+sed 's|diag libinstaller dos win32 win64 dosutil txt|libinstaller txt|g' -i Makefile
+sed 's|win32/syslinux.exe win64/syslinux64.exe||g' -i Makefile
+sed 's|dosutil/\*.com dosutil/\*.sys||g' -i Makefile
+sed 's|dos/syslinux.com||g' -i Makefile
+sed 's|INSTALLSUBDIRS = com32 utils dosutil|INSTALLSUBDIRS = com32 utils|g' -i Makefile
+sed "s|install -m 644 -c $(INSTALL_DIAG) $(INSTALLROOT)$(DIAGDIR)|# install -m 644 -c $(INSTALL_DIAG) $(INSTALLROOT)$(DIAGDIR)|g" -i Makefile
+
+# fix incorrect manpath
+sed 's|/usr/man|/usr/share/man|g' -i mk/syslinux.mk
+
```
Comment (dylanaraps) [inline community/syslinux/build]: Add `: > mk/devel.mk` to disable debug and development flags. Will reduce the overall size.

**Comment by dylanaraps** [comment]
Missing make dependency on `python` and `nasm`?

File: community/syslinux/build [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# remove Windows related...
+sed 's|diag libinstaller dos win32 win64 dosutil txt|libinstaller txt|g' -i Makefile
+sed 's|win32/syslinux.exe win64/syslinux64.exe||g' -i Makefile
+sed 's|dosutil/\*.com dosutil/\*.sys||g' -i Makefile
+sed 's|dos/syslinux.com||g' -i Makefile
+sed 's|INSTALLSUBDIRS = com32 utils dosutil|INSTALLSUBDIRS = com32 utils|g' -i Makefile
+sed "s|install -m 644 -c $(INSTALL_DIAG) $(INSTALLROOT)$(DIAGDIR)|# install -m 644 -c $(INSTALL_DIAG) $(INSTALLROOT)$(DIAGDIR)|g" -i Makefile
+
+# fix incorrect manpath
+sed 's|/usr/man|/usr/share/man|g' -i mk/syslinux.mk
```
Comment (dylanaraps) [inline community/syslinux/build]: Just add `MANDIR=/usr/share/man` to `make`. :+1:

File: community/syslinux/build [hunk 6]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh -e
+
+unset CFLAGS LDFLAGS
+
+for f in *.patch ; do patch -p1 < "$f"; done
```
Comment (dylanaraps) [inline community/syslinux/build]: Replace the entire patch with:

```sh
# Fix missing <sys/sysmacros.h> include.
sed -i 's@vfs.h>@vfs.h>\n#include <sys/sysmacros.h>@' extlinux/main.c
```

**Comment by dylanaraps** [comment]
```sh
#!/bin/sh -e

unset CFLAGS LDFLAGS

# Fix missing <sys/sysmacros.h> include.
sed -i 's@vfs.h>@vfs.h>\n#include <sys/sysmacros.h>@' extlinux/main.c

# Disable debugging and development "stuff".
: > mk/devel.mk

make clean

make \
    PYTHON=python \
    bios \
    efi64 \
    installer

make -j1 \
    MANDIR=/usr/share/man \
    INSTALLROOT="$1" \
    bios \
    efi64 \
    install
```

**Comment by dylanaraps** [comment]
Source: https://mirrors.edge.kernel.org/pub/linux/utils/boot/syslinux/Testing/6.04/syslinux-6.04-pre1.tar.xz

**Comment by dylanaraps** [comment]
Dependencies: 

```
binutils make
nasm     make
perl     make
python   make
util-linux
```

**Comment by a-schaefers** [comment]
This should be our recommended legacy bios boot loader :) It is very stupid simple to use (after you can get it to compile. Lol)

**Comment by dylanaraps** [comment]
Shall we close this?

**Comment by a-schaefers** [comment]
I don't want it. ¯\_(ツ)_/¯

**Pull request closed without merging**